modules/aggregation-layers/src/aggregation-layer.js

```diff
@@ -18,24 +18,25 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-import {CompositeLayer, AttributeManager, _compareProps as compareProps} from '@deck.gl/core';
+import {
+  Layer,
+  CompositeLayer,
+  AttributeManager,
+  _compareProps as compareProps
+} from '@deck.gl/core';
 import {cssToDeviceRatio} from '@luma.gl/core';
-
-// props when changed results in new uniforms that requires re-aggregation
-const UNIFORM_PROPS = [
-  // DATA-FILTER extension
-  'filterEnabled',
-  'filterRange',
-  'filterSoftRange',
-  'filterTransformSize',
-  'filterTransformColor'
-];
+import {filterProps} from './utils/prop-utils';
 
 export default class AggregationLayer extends CompositeLayer {
-  initializeState(aggregationProps = []) {
+  initializeState(dimensions) {
     super.initializeState();
+
+    // props , when changed doesn't require updating aggregation
+    const ignoreProps = filterProps(this.constructor._propTypes, dimensions.data.props);
+
     this.setState({
-      aggregationProps: aggregationProps.concat(UNIFORM_PROPS)
+      ignoreProps: Object.assign(ignoreProps, Layer.defaultProps),
```

Comment:
Layer.defaultProps not needed.

Reply:
If updateTrigger[key] and key is not in `ignoreProps`  - trigger re-aggreagtion (for GPU only). => attributeChanged is enough.